blog/2024-12-15-openhab-4-3-release.md

```diff
@@ -0,0 +1,294 @@
+---
+layout: BlogPost
+category: blog
+title: openHAB 4.3 Release
+author: Florian Hotze and others
+date: "2024-12-15T18:00:00+01:00"
+previewimage: /uploads/2024-12-15-openhab-4-3-release/header.png
+tags:
+  - news
+excerpt: >-
+  We are pleased to announce the release of openHAB 4.3!
+  This traditional winter release introduces numerous new features and enhancements to the openHAB ecosystem, paving the way for our next major release.
+---
+
+We are pleased to announce the release of openHAB 4.3!
+This traditional winter release introduces numerous new features and enhancements to the openHAB ecosystem, paving the way for our next major release.
+
+One and a half year have passed since the [release of the openHAB 4.x series](/blog/2023-07-23-openhab-4-0-release.html), and we are proud to announce the final version of the openHAB 4.x series today.
+However, openHAB 4.3 is far from being just a minor step toward openHAB 5 — it is packed with incredible new features, including a log viewer, UI support for Thing actions, and the new interactive SVG canvas, which unlocks previously untapped possibilities for your smart home dashboard.
+Having openHAB 4.3 as a solid foundation, we will now focus development on openHAB 5, for which we have already started to collect ideas.
+Make sure to read the outlook towards openHAB 5, it contains some important news!
+
+That being said, let's check out some statistics praising the activity of our community and dive into the highlights of the openHAB 4.3 release.
+
+# Activity
+
+# Highlights
+
+As you might have already noticed in the activity statistics, there was much development on openHAB - too much to list everything.
+Please refer to the [**release notes**](https://github.com/openhab/openhab-distro/releases/tag/4.3.0) to check out what's new and noteworthy (e.g. breaking changes).
+
+In the following sections, our maintainers and contributors are introducing some new features you don't want to miss!
+
+## Core Runtime Enhancements
+
+### Pre-Compile Script Conditions and Actions
+
+_Florian Hotze ([@florian-h05](https://github.com/florian-h05)), openHAB Maintainer_
+
+openHAB 4.3 introduces automatic pre-compilation of script conditions and actions from UI-based rules.
+
+This means that these scripts are now compiled on openHAB startup instead of on the first execution and there is no delay any more when the rule is executed for the first time.
+Another benefit is that syntax errors are discovered at startup and not at runtime of the rule. Makes it easier to find and fix them.
+
+Especially Blockly users on Raspberry Pis with a 32-bit OS will love this change, as the JavaScript Scripting automation add-on, which is used by Blockly under the hood, is known to compile scripts very slowly on these devices.
+Please note that not all automation add-ons support pre-compilation (yet).
+
```

Comment:
Here is some text for persistence item and group exclusion:

```suggestion

### Allow excluding Items and Groups from persistence

You may want to persist all items but a few.
While this has always been possible by defining a group containing all items to persist and only persisting the items in this group, it creates a maintenance burden to always have to add new items you want to persist to this group.

The new exclusion functionality now allows you to easily exclude individual items or groups of items from a persistence strategy and service. This can be configured in persist files by putting a ! in front of the items or groups to exclude, or through the UI persistence configuration.
```